### PR TITLE
[INLONG-5092][SDK] Change Kafka default partition assignment strategy to RangeAssignor

### LINKED ISSUE #5092: [Feature][SortSdk] Change Kafka default partition assignment strategy to RangeAssignor
### Description

Change Kafka default partition assignment strategy to RangeAssignor.
The partition assignment strategy of StickyAssignor will make more duplicated data when the cluster node of SortStandalone.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luchunliang**
### Prepare a Pull Request
- Title: [INLONG-5092][SortSdk]Change Kafka default partition assignment strategy to RangeAssignor
- Fixes #5092 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by vernedeng — APPROVED** [review]
LGTM

**Comment by dockerzhang** [comment]
@yunqingmoswu @gong PTAL, thanks.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsarzone/PulsarClusterProducer.java [hunk 1]
```diff
@@ -282,6 +281,27 @@ public boolean send(DispatchProfile event) {
         }
     }
 
+    /**
+     * getProducerTopic
+     * @param event
+     * @return
```
Comment (EMsnap) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsarzone/PulsarClusterProducer.java:287]: no description for return

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java [hunk 2]
```diff
@@ -19,43 +19,43 @@
...
+    public AckOffsetOnRebalance(String clusterId, KafkaConsumer<byte[], byte[]> consumer,
             ConcurrentHashMap<TopicPartition, OffsetAndMetadata> commitOffsetMap) {
+        this.clusterId = clusterId;
         this.consumer = consumer;
         this.commitOffsetMap = commitOffsetMap;
     }
 
     @Override
     public void onPartitionsRevoked(Collection<TopicPartition> collection) {
         logger.debug("*- in ralance:onPartitionsRevoked");
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java:48]: The log info should be standardized, and the use of special characters is not recommended.
```java
logger.debug("execute Rebalance:onPartitionsRevoked");
```

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java [hunk 3]
```diff
@@ -19,43 +19,43 @@
...
-            consumer.commitSync(commitOffsetMap);
-        }
+        collection.forEach((v) -> {
+            logger.info("clusterId:{},onPartitionsRevoked:{}", clusterId, v.toString());
+        });
     }
 
     @Override
     public void onPartitionsAssigned(Collection<TopicPartition> collection) {
         logger.debug("*- in ralance:onPartitionsAssigned  ");
```
Comment (healchow) [inline inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java:56]: The log info should be standardized, and the use of special characters is not recommended.
```java
logger.debug("execute Rebalance:onPartitionsAssigned");
```

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsarzone/PulsarClusterProducer.java (hunk 1 above)
Comment (luchunliang) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsarzone/PulsarClusterProducer.java]: add description

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java (hunk 2 above)
Comment (luchunliang) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java]: fixed

File: inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java (hunk 3 above)
Comment (luchunliang) [reply inlong-sdk/sort-sdk/src/main/java/org/apache/inlong/sdk/sort/impl/kafka/AckOffsetOnRebalance.java]: fixed

**Review by healchow — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**